### PR TITLE
Boleto component

### PR DESCRIPTION
**Pull request by etman55**
## Checklist 
- [x] Code is unit tested
- [x] Changes are tested manually

image  image

COAND-366

### CODE REVIEW

File: components-core/src/main/java/com/adyen/checkout/components/core/PaymentMethodTypes.kt [hunk 1]
```diff
@@ -94,6 +95,13 @@ object PaymentMethodTypes {
         listOf(
             ACH,
             BCMC,
+            BOLETOBANCARIO,
```
Comment (OscarSpruit) [inline components-core/src/main/java/com/adyen/checkout/components/core/PaymentMethodTypes.kt:98]: With this PR Boleto is not fully supported yet, would be better to leave it in the unsupported list and then add them here with the voucher PR

**Comment by jreij** [comment]
We're missing configuration from Boleto in `DropInConfiguration`

File: ui-core/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <dimen name="standard_half_margin">8dp</dimen>
     <dimen name="standard_three_quarters_margin">12dp</dimen>
     <dimen name="standard_margin">16dp</dimen>
+    <dimen name="standard_one_half_margin">24dp</dimen>
```
[the remark this answers is not in the document]
Comment (etman55) [reply ui-core/src/main/res/values/dimens.xml:16]: this was suppose to be for voucher view and it got here somehow 😅

File: boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/view/BoletoView.kt [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            localizedContext
+        )
+        binding.textInputLayoutShopperEmail.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Boleto_ShopperEmailInput,
+            localizedContext
+        )
+    }
+
+    private fun initFirstNameInput() {
+        val firstNameEditText = binding.textInputLayoutFirstName.editText as? AdyenTextInputEditText
```
[the remark this answers is not in the document]
Comment (etman55) [reply boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/view/BoletoView.kt:146]: as `setOnChangeListener()` is not an extension function.

File: ui-core/src/main/res/values-ar/strings.xml [hunk 4]
```diff
@@ -42,4 +43,5 @@
     <string name="checkout_address_form_province_territory_hint_optional">المقاطعة أو الإقليم (اختياري)</string>
     <string name="checkout_address_form_city_town_hint_optional">المدينة / البلدة (اختياري)</string>
 
+
```
[the remark this answers is not in the document]
Comment (etman55) [reply ui-core/src/main/res/values-ar/strings.xml:46]: hmm, I think it was added by mistake after commit check.

File: boleto/src/test/java/com/adyen/checkout/boleto/internal/ui/BoletoDelegateTest.kt [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.verify
+import java.util.Locale
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(MockitoExtension::class)
+internal class BoletoDelegateTest(
```
Comment (jreij) [inline boleto/src/test/java/com/adyen/checkout/boleto/internal/ui/BoletoDelegateTest.kt:47]: Should this be called `DefaultBoletoDelegateTest`?

File: drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt [hunk 6]
```diff
@@ -514,6 +523,16 @@ internal fun getComponentFor(
                 callback = componentCallback as ComponentCallback<BlikComponentState>,
             )
         }
+        BoletoComponent.PROVIDER.isPaymentMethodSupported(paymentMethod) -> {
+            val upiConfiguration: BoletoConfiguration =
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt:527]: Should be named `boletoConfiguration`

File: boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/view/BoletoView.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/view/BoletoView.kt:146]: Hmm not sure what you mean, but it's possible to do `binding.editTextFirstName.setOnChangeListener { ... }`

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt [hunk 7]
```diff
@@ -346,8 +347,20 @@ class DropInConfiguration private constructor(
          */
         fun addUPIConfiguration(upiConfiguration: UPIConfiguration): Builder {
             availablePaymentConfigs[PaymentMethodTypes.UPI] = upiConfiguration
-            availablePaymentConfigs[PaymentMethodTypes.UPI_COLLECT] = upiConfiguration
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt:349]: This is not supposed to be deleted

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ViewExtensions.kt [hunk 8]
```diff
@@ -21,7 +21,6 @@ inline var TextInputLayout.isVisible: Boolean
         this.visibility = visibility
         editText?.apply {
             this.visibility = visibility
-            isFocusable = value
```
Comment (OscarSpruit) [inline ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ViewExtensions.kt:24]: Why was this removed?

File: boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/DefaultBoletoDelegate.kt [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+        if (AddressFormUtils.isAddressRequired(outputData.addressUIState)) {
+            paymentComponentData.billingAddress = AddressFormUtils.makeAddressData(
+                addressOutputData = outputData.addressState,
+                addressFormUIState = outputData.addressUIState
+            )
+        }
+        return BoletoComponentState(
+            data = paymentComponentData,
+            isInputValid = outputData.isValid,
+            isReady = true
```
Comment (OscarSpruit) [inline boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/DefaultBoletoDelegate.kt:255]: Same thing as before: are we sure the component is finished with loading the address data?

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ViewExtensions.kt (hunk 8 above)
Comment (etman55) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ViewExtensions.kt:24]: as if you changed the visibility of the email view more than one time you can't focus anymore on it and it has nothing to do with changing the visibility of the view.

Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ViewExtensions.kt:24]: It looks like it was added on purpose. @jreij do you remember why this was added?

File: boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/DefaultBoletoDelegate.kt [hunk 10]
```diff
@@ -234,6 +235,7 @@ internal class DefaultBoletoDelegate(
         val paymentComponentData = PaymentComponentData(
             paymentMethod = GenericPaymentMethod(paymentMethod.type),
             order = order,
+            amount = componentParams.amount.takeUnless { it.isEmpty },
```
Comment (jreij) [inline boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/DefaultBoletoDelegate.kt:239]: Don't forget to add tests for this, they can be copied from other modules

**Review by jreij — APPROVED** [review]

**Review by jreij — APPROVED** [review]
